seismicpro/src/seismic_metrics.py

```diff
@@ -0,0 +1,135 @@
+"""File contains metircs for seismic processing."""
+import numpy as np
+from numba import njit, prange
+
+from ..batchflow.models.metrics import Metrics
+
+from .plot_utils import plot_metrics_map
+
+
+class MetricsMap(Metrics):
+    """seismic metrics class"""
+
+    def __init__(self, metrics, coords, *args, **kwargs):
+        _ = args, kwargs
+        super().__init__()
+
+        self._metrics_values = list(metrics) if isinstance(metrics, (list, tuple, set, np.ndarray)) else [metrics]
+        self._coords = list(coords)
+
+        if len(self._metrics_values) != len(self._coords):
+            raise ValueError('length of given metrics do not match with length of coords.')
+
+        self._agg_fn_dict = {'mean': np.nanmean,
+                             'max': np.nanmax,
+                             'min': np.nanmin}
+
+    @property
+    def metrics(self):
+        """Accumulated metrics values. """
+        return self._metrics_values
+
+    @property
+    def coords(self):
+        """Accumulated coordinates. """
+        return self._coords
+
+
+    def extend(self, metrics):
+        """Extend coordinates and metrics to global container."""
+        self._metrics_values.extend(metrics.metrics_values)
+        self._coords.extend(metrics.coords)
+
+    def construct_map(self, bin_size=500, cm=None, title=None, figsize=None, save_to=None, dpi=None, #pylint: disable=too-many-arguments
+                      pad=False, plot=True, agg_bins_fn='mean', agg_bins_kwargs=None, **plot_kwargs):
+        """ Each value in resulted map represent aggregated value of metrics for coordinates belongs to current bin.
+        """
+
+        if isinstance(bin_size, int):
+            bin_size = (bin_size, bin_size)
+
+        coords = np.array(self._coords)
+        coords_x = np.array(coords[:, 0], dtype=np.float32)
+        coords_y = np.array(coords[:, 1], dtype=np.float32)
+
+        metrics = np.array(list(self._metrics_values))
+
+        if isinstance(metrics[0], (tuple, list, set, np.ndarray)):
+            if len(np.array(metrics[0]).shape) > 1:
+                raise ValueError('Construct map does not work with 3d metrics yet.')
+            metrics_shapes = np.array([metric.shape for metric in metrics])
+            metrics_storage = np.empty((len(metrics), np.max(metrics_shapes)))
+            metrics_storage.fill(np.nan)
+            mask = np.zeros_like(metrics_storage) + np.arange(np.max(metrics_shapes))
+            mask = mask < metrics_shapes
+            metrics_storage[mask] = np.concatenate(metrics)
+            metrics = metrics_storage.copy()
+        metrics = np.array(metrics, dtype=np.float32)
+
+        if isinstance(agg_bins_fn, str):
+            call_agg_bins = getattr(NumbaNumpy, agg_bins_fn)
+            call_agg_bins = call_agg_bins(None, **agg_bins_kwargs) if agg_bins_kwargs is not None else call_agg_bins
```

Comment:
```suggestion
            call_agg_bins = call_agg_bins(**agg_bins_kwargs) if agg_bins_kwargs else call_agg_bins
```
see comment to line 134
if `agg_bins_kwargs` is an empty dict, you shouldn't pass it to `call_agg_bins`, should you?